### PR TITLE
[ADD] Inventory: Add Replenish on order (MTO)

### PR DESCRIPTION
**Pull request by hojo-odoo**
This PR adds a new doc for the Inventory app detailing how to use the Replenish on Order (MTO) route. 

This doc expands upon the information in the MTO subsection of the [How to select the right replenishment strategy](https://www.odoo.com/documentation/14.0/applications/inventory_and_mrp/inventory/management/products/strategies.html) doc, which also contains subsections about replenishment rules and the Replenishment page. Two additional docs will be pushed regarding those topics, at which point the "How to select the right replenishment strategy" doc will be deleted.

### CODE REVIEW

File: content/applications/inventory_and_mrp/inventory/management/products/mto.rst [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+.. image:: mto/specify-vendor.png
+   :align: center
+   :alt: Enable "Can be Purchased" and specify a vendor.
+
+If the product will be manufactured, click the :guilabel:`Bill of Materials` smart button at the top
+of the screen, then :guilabel:`Create` to configure a new Bill of Materials for the product. Add the
+components used to manufacture the product on the :guilabel:`Components` tab, along with the
+operations required for the manufacturing workflow on the :guilabel:`Operations` tab. Finally, click
+:guilabel:`Save` to save the Bill of Materials.
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/management/products/mto.rst:74]:
```suggestion
If the product is manufactured, click the :guilabel:`Bill of Materials` smart button at the top of
the screen, then click :guilabel:`Create` on the :guilabel:`Bill of Materials` page to configure a
new Bill of Materials for the product. Doing so reveals a blank Bill of Materials form.

On the blank Bill of Materials form, add the components used to manufacture the product under the
:guilabel:`Components` tab. In the along with the :guilabel:`Operations` tab, add operations
required for the manufacturing workflow, by clicking :guilabel:`Add a line`, and entering in the
necessary information in the :guilabel:`Create Operations` pop-up.

Then, click :guilabel:`Save & Close` to save and return to the Bill of Materials form, or
:guilabel:`Save & New` to save and immediately create another operation on a fresh
:guilabel:`Create Operations` pop-up.

Lastly, when all necessary configurations are complete, click :guilabel:`Save` to save the Bill of
Materials.
```

File: content/applications/inventory_and_mrp/inventory/management/products/mto.rst [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+Fulfill a Sales Order using the MTO route
+=========================================
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/management/products/mto.rst:77]:
```suggestion
Fulfill sales order with MTO route
==================================
```

Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/management/products/mto.rst:77]:
```suggestion
Fulfill sales order with MTO route
==================================
```

File: content/applications/inventory_and_mrp/inventory/management/products/mto.rst [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+After configuring a product to use the MTO route, a replenishment order will be created for it every
+time a Sales Order including the product is confirmed. The type of order created depends on the
+second route selected in addition to MTO. For example, if *Buy* was the second route
+selected, then a Purchase Order will be created upon confirmation of a Sales Order.
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/management/products/mto.rst:82]:
```suggestion
After configuring a product to use the MTO route, a replenishment order is created for it every
time a Sales Order including that specific product is confirmed. The type of order created depends
on the second route selected, in addition to MTO.

For example, if *Buy* was the second route selected on the product form, then a Purchase Order is
created upon confirmation of a Sales Order.
```

File: content/applications/inventory_and_mrp/inventory/management/products/mto.rst [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+.. important::
+   When the MTO route is enabled for a product, a replenishment order will always be created upon
+   confirmation of a Sales Order. This is the case even if there is enough stock of the product on
+   hand to fulfill the Sales Order without buying or manufacturing additional units of it.
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/management/products/mto.rst:87]:
```suggestion
   When the MTO route is enabled for a product, a Replenishment Order is always created upon
   confirmation of a Sales Order. This is the case even if there is enough stock of the product
   on-hand to fulfill the Sales Order, without buying or manufacturing additional units of it.
```

File: content/applications/inventory_and_mrp/inventory/management/products/mto.rst [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+While the MTO route can be used in unison with a variety of other routes, the *Buy* route
+will be used as the example for this workflow. Begin by navigating to the :guilabel:`Sales` app,
+then click :guilabel:`Create` to create a new Quotation.
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/management/products/mto.rst:91]:
```suggestion
While the MTO route can be used in unison with a variety of other routes, the *Buy* route
is used as the example for this workflow. Begin by navigating to the :menuselection:`Sales` app,
then click :guilabel:`Create`, which reveals a blank Quotation form.
```

File: content/applications/inventory_and_mrp/inventory/management/products/mto.rst [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+Add a :guilabel:`Customer` to the Quotation, then click :guilabel:`Add a product` under the
+:guilabel:`Order Lines` tab and enter a product that has been configured to use the MTO and Buy
+routes. Click :guilabel:`Confirm` and the Quotation will be turned into a Sales Order.
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/management/products/mto.rst:95]:
```suggestion
On the blank Quotation form, add a :guilabel:`Customer`, then click :guilabel:`Add a product` under
the :guilabel:`Order Lines` tab, and enter a product that has been configured to use the *MTO* and
*Buy* routes. Then, click :guilabel:`Confirm`, and the Quotation is turned into a Sales Order.
```

File: content/applications/inventory_and_mrp/inventory/management/products/mto.rst [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+A :guilabel:`Purchase` smart button will now appear in the top right corner of the Sales Order.
+Click it to be taken to the associated Request for Quotation. Click :guilabel:`Confirm Order` to
+confirm the Request for Quotation and turn it into a Purchase Order. A green :guilabel:`Receive
+Products` button will now appear at the top of the Purchase Order. Once the products are received,
+click it to register them into inventory.
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/management/products/mto.rst:101]:
```suggestion
A :guilabel:`Purchase` smart button appears in the top-right corner of the Sales Order. Clicking it
reveals the Request for Quotation associated with that specific Sales Order.

Click :guilabel:`Confirm Order` to confirm the Request for Quotation, and turn it into a Purchase
Order. Doing so reveals a green :guilabel:`Receive Products` button at the top of the Purchase
Order. Click that button once the products are received to register them into inventory.
```

File: content/applications/inventory_and_mrp/inventory/management/products/mto.rst [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+
+Return to the Sales Order by clicking the SO breadcrumb or by navigating to :menuselection:`Sales
+--> Orders --> Orders` and selecting the Sales Order. Click the :guilabel:`Delivery` smart button in
+the top right of the order to be taken to the delivery order. Once the products have been shipped to
+the customer, click :guilabel:`Validate` to confirm the delivery.
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/management/products/mto.rst:106]:
```suggestion
Return to the Sales Order by clicking the :guilabel:`SO` breadcrumb, or by navigating to
:menuselection:`Sales app --> Orders --> Orders` and selecting the desired Sales Order.

Then, click the :guilabel:`Delivery` smart button in the top-right of the order to be taken to the
Delivery Order. Once the products have been shipped to the customer, click :guilabel:`Validate` to
confirm the delivery.
```

**Review by ksc-odoo — COMMENTED** [review]
@hojo-odoo just finished reviewing this doc. Fantastic job. Just a few minor adjustments, but it should be good to go after that. If you have any questions or need clarification on anything, just let me know. Once again, great work! 👍

**Review by ksc-odoo — COMMENTED** [review]
@hojo-odoo just finished reviewing this doc. Fantastic job. Just a few minor adjustments, but it should be good to go after that. If you have any questions or need clarification on anything, just let me know. Once again, great work! 👍

File: content/applications/inventory_and_mrp/inventory/management/products/mto.rst [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+:guilabel:`Operations` tab.
+
+Finally, click :guilabel:`Save` to save the Bill of Materials.
+
+Fulfill a Sales Order using the MTO route
+=========================================
+
+After configuring a product to use the MTO route, a replenishment order is created for it every time
+a Sales Order including the product is confirmed. The type of order created depends on the second
+route selected in addition to MTO.
```
Comment (samueljlieber) [inline content/applications/inventory_and_mrp/inventory/management/products/mto.rst:87]: Adding abbreviations and no caps on _sales order_
```suggestion
After configuring a product to use the |MTO| route, a replenishment order is created for it every
time a sales order including the product is confirmed. The type of order created depends on the
second route selected in addition to |MTO|.
```

File: content/applications/inventory_and_mrp/inventory/management/products/mto.rst [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+
+For example, if *Buy* was the second route selected, then a Purchase Order is created upon
+confirmation of a Sales Order.
```
Comment (samueljlieber) [inline content/applications/inventory_and_mrp/inventory/management/products/mto.rst:90]: Remove caps
```suggestion
For example, if *Buy* was the second route selected, then a purchase order is created upon
confirmation of a sales order.
```

**Review by samueljlieber — CHANGES_REQUESTED** [review]
Hi @hojo-odoo ✌️ great job with this new doc! It is looking really great, I have just a few technical changes– most are removing capitalization from general industry terms. 

You will a suggestion for adding abbreviation shorthands, let me know if you have any questions on that!

Thank you for the continued hard work on this PR 🙏

**Comment by hojo-odoo** [comment]
Hey @samueljlieber,

This should be ready for Zach too. I addressed all of your comments.

Thanks,
hojo

File: content/applications/inventory_and_mrp/inventory/management/products/mto.rst [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+:guilabel:`Create` on the :guilabel:`Bill of Materials` page to configure a new |BOM| for the
+product.
+
+On the blank |BOM| form, add the components used to manufacture the product on the
+:guilabel:`Components` tab, along with the operations required for the manufacturing workflow on the
+:guilabel:`Operations` tab.
+
+Finally, click :guilabel:`Save` to save the |BOM|.
+
+Fulfill a Sales Order using the MTO route
```
Comment (samueljlieber) [inline content/applications/inventory_and_mrp/inventory/management/products/mto.rst:89]: No caps needed on _sales order_
```suggestion
Fulfill a sales order using the MTO route
```

**Review by samueljlieber — APPROVED** [review]
Hi @hojo-odoo great job with this PR! It is looking great to me 🙂 I am approving now but please make the changes below before passing to final review!

File: content/applications/inventory_and_mrp/inventory/management/products/mto.rst [hunk 12]
```diff
@@ -0,0 +1,122 @@
+========================
+Replenish on order (MTO)
+========================
+
+.. |MTO| replace:: :abbr:`MTO (make to order)`
+.. |RFQ| replace:: :abbr:`RFQ (request for quotation)`
+.. |BOM| replace:: :abbr:`BOM (bill of materials)`
```
Comment (StraubCreative) [inline content/applications/inventory_and_mrp/inventory/management/products/mto.rst:7]: This is how we've been formatting the capitalization of these acronyms.
```suggestion
.. |MTO| replace:: :abbr:`MTO (Make to Order)`
.. |RFQ| replace:: :abbr:`RFQ (Request for Quotation)`
.. |BOM| replace:: :abbr:`BoM (Bill of Materials)`
```

**Review by StraubCreative — APPROVED** [review]
Looks good @hojo-odoo 

On future versions I think it would be valuable to also include an MTO + Manufacture route section with a full walk-through of managing an MTO-manufactured product.

This should get us started though, as it's immediately useful for customers 😉 
Don't forget to update 16.0 with new UI screenshots and remove "save" and/or "edit" language where necessary.

@robodoo r+

**Pull request closed without merging**